Uprev to GHC 8.10.7

crogers1 opened the pull request:
Refer to individual commits for more details. The primary PR can be found against xenclient-oe here:

https://github.com/OpenXT/xenclient-oe/pull/1469

Commit:
[ghc8] Refresh Setup.hs files

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Deprecate haskell98

This includes "import System", which is no longer valid.

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Update Directory import to System.Directory

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Derive Applicative where needed

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Prevent ghc from complaining about tabs

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Convert Text to String

The haskell-dbus library now uses Strings for parsing.
Note that we are also deprecating a helper method called
objPathToStr as it is no longer necessary.

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] Add missing language pragmas

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] nws: Replace USER with User

As per the new hsyslog API.

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] nwd: Add explicit typing to varToArrStr fn

This function will split out a slew of "ambiguous reference"
errors as the compiler can't deduce the return type for chained
operations.

Add the explicit typing and remove the seemingly equivalent (but
never used) networkObjPaths function.

Signed-off-by: Nicholas Tsirakis <tsirakisn@ainfosec.com>

Commit:
[ghc8] nws: Just call ip directly

  There's no need to pass the whole path. And ghc8 will error
  in createProcess since it can't find /bin/ip (since the actual
  path is /sbin/ip)

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

jandryuk reviewed (changes requested):
Looks good.  Could you please add a blurb to [[ghc8] nwd: nws: Don't use RTS anymore](https://github.com/OpenXT/network/pull/27/commits/ce1535533387deb380d57151cc7ff7dbf837a399) stating why RTS is dropped and what it means?  Thanks

Commit:
[ghc8] nws: nwd: Use rtsopts flag during build

  This allows us to pass the RTS flag in the initscripts for
  network-daemon and slave.

  Additionally, add colon after the "GHC-Options" section specifier.
  This has been missing since 2014, using the colon here should properly
  set the options we're expecting.

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

crogers1 commented:
Dropped the "Dont use RTS anymore" commit, it's not needed.  I simply forgot `-rtsopts` in the cabal files.  The initiscripts have been restored to their original versions with the `+RTS` opts to set max generations and generation size.
